README.md

```diff
@@ -173,33 +173,85 @@ In case you have multiple hierarchies that reuse the same enum TypeProvider can
 In case you don't want to (or can't - third party API) include type information in json, you can use this approach:
 
 ```java
-interface FooBar extends PresentPropertyJsonHierarchy<FooBarType> {
+interface Bike extends PresentPropertyJsonHierarchy<BikeType> {
 }
 
 @AllArgsConstructor(onConstructor_ = @JsonCreator)
 @Value
-static class Foo implements FooBar {
-    private final String foo;
+static class RoadBike implements Bike {
+    private final String roadBike;
 }
 
 @AllArgsConstructor(onConstructor_ = @JsonCreator)
 @Value
-static class Bar implements FooBar {
-    private final String bar;
+static class Bmx implements Bike {
+    private final String bmx;
 }
 
 @Getter
 @AllArgsConstructor
-enum FooBarType implements TypeProvider {
-    FOO(Foo.class),
-    BAR(Bar.class);
+enum BikeType implements TypeProvider {
+    ROAD_BIKE(RoadBike.class),
+    BMX(Bmx.class);
 
-    private final Class<? extends FooBar> type;
+    private final Class<? extends Bike> type;
 }
 ```
 
 [Showcase](infobip-jackson-extension-module/src/test/java/com/infobip/jackson/PresentPropertyDeserializerTest.java).
 
+Notice that by default snake cased type names are translated to camel case properties (e.g. `ROAD_BIKE` -> `roadBike`).
+If you want to use different casing you can provide your own resolver by extending `PresentPropertyJsonTypeResolver`:
+```java
+static class LowerUnderscorePresentPropertyJsonTypeResolver<E extends Enum<E> & TypeProvider> extends PresentPropertyJsonTypeResolver<E> {
+
+    public LowerUnderscorePresentPropertyJsonTypeResolver(Class<E> type) {
+        super(type, CaseFormat.LOWER_UNDERSCORE);
+    }
+}
+```
+Then your model may look as follows:
+```java
+@JsonTypeResolveWith(LowerUnderscorePresentPropertyJsonTypeResolver.class)
+interface Bike extends PresentPropertyJsonHierarchy<BikeType> {
+
+}
+
+@AllArgsConstructor(onConstructor_ = @JsonCreator)
+@Value
+static class RoadBike implements Bike {
+
+    private final String road_bike;
```

Comment:
it's never a convention to use snakeCase in field names, you should use `@JsonNaming` instead

Reply:
Fixed